src/_algopy_testing/decorators/subroutine.py

```diff
@@ -1,9 +1,26 @@
+import functools
+import typing
 from collections.abc import Callable
 from typing import ParamSpec, TypeVar
```

Comment:
purely stylistic, but I prefer not to have both `from x import ...` and `import x` in the same file, suggest changing this existing import to work with `import x`

Reply:
It has been updated to use `from x import ...` format consistently.